tests/Functional/Net/SFTPUserStoryTest.php

```diff
@@ -649,5 +649,22 @@ public function testStatVsLstat($sftp)
         $this->assertSame($stat['type'], NET_SFTP_TYPE_SYMLINK);
 
         $sftp->enableStatCache();
+
+        return $sftp;
+    }
+
+    /**
+     * @depends testStatVsLstat
+     * @group github830
+     */
+    public function testEndlessLoopOnUpload($sftp)
```

Comment:
The second hunk is already there and no one has created a bug report for it. It's possible people like the current behavior even tho it, imho, doesn't mesh with the documentation. It could be changed but idk...  atm it's not a priority of mine.